examples/bull-spread/bull-spread.go

```diff
@@ -0,0 +1,161 @@
+)
+
+var decimal10k = decimal.NewFromInt(10000)
+
+const (
+	oneWeek = 7
+)
+
+type Service struct {
+	tdClient *alpaca.Client
+	mdClient *marketdata.Client
+	acct     *alpaca.Account
+}
+
+func NewService() (*Service, error) {
+	// You can set your API key/secret here or you can use environment variables
+	apiKey := os.Getenv("APCA_API_KEY_ID")
+	apiSecret := os.Getenv("APCA_API_SECRET_KEY")
+	// NOTE: mleg complex option strategies are still in beta only availabe in Paper
+	baseURL := "https://paper-api.alpaca.markets"
+
+	tdClient := alpaca.NewClient(alpaca.ClientOpts{
+		APIKey:    apiKey,
+		APISecret: apiSecret,
+		BaseURL:   baseURL,
+	})
+
+	mdClient := marketdata.NewClient(marketdata.ClientOpts{
+		APIKey:    apiKey,
+		APISecret: apiSecret,
+	})
+
+	// Cancel any open orders so they don't interfere with this algo
+	if err := tdClient.CancelAllOrders(); err != nil {
+		return nil, err
+	}
+
+	// Make sure we have enough green for some mleg fun
+	acct, err := tdClient.GetAccount()
+	if err != nil {
+		return nil, err
+	}
+	if acct.BuyingPower.LessThan(decimal10k) {
+		return nil, fmt.Errorf("insufficient buying power: needed %s, have %s",
+			decimal10k.String(), acct.BuyingPower.String())
+	}
+
+	service := Service{tdClient: tdClient, mdClient: mdClient, acct: acct}
+	return &service, nil
+}
+
+func main() {
+	svc, err := NewService()
+	if err != nil {
+		log.Fatalf("failed to intilize service: %v", err)
+	}
+
+	underlying := "INTC"
+	td, err := svc.mdClient.GetLatestTrade(underlying, marketdata.GetLatestTradeRequest{Feed: "iex"})
+	if err != nil {
+		log.Fatalf("getting latest trade for symbol %s: %v", underlying, err)
+	}
+	px := td.Price
+
+	// list contracts that are expiring at least a week from now and strike is GTE latest trade
+	now := time.Now()
+	dt := civil.DateOf(time.Date(now.Year(), now.Month(), now.Day()+oneWeek, 0, 0, 0, 0, time.UTC))
+
+	// Select two contracts for the two legs of the spread:
+	// 1. long leg, strike A at latest trade
+	var (
+		long  alpaca.OptionContract
+		short alpaca.OptionContract
+	)
+	{
+		req := alpaca.GetOptionContractsRequest{
+			UnderlyingSymbols: underlying,
+			Status:            alpaca.OptionStatusActive,
+			ExpirationDateGTE: dt,
+			StrikePriceGTE:    decimal.NewFromFloat(px), // strike A
+			TotalLimit:        1,
+		}
+		contracts, err := svc.tdClient.GetOptionContracts(req)
+		if err != nil {
+			log.Fatalf("listing contracts: %v", err)
+		}
+		if len(contracts) < 1 {
+			log.Fatal("no contracts to choose from for the long leg")
+		}
+		long = contracts[0]
+		log.Printf("-> contract to be bought long: %s", long.Symbol)
+	}
+	// 2. short leg, strike B at $10 above latest trade
+	{
+		req := alpaca.GetOptionContractsRequest{
+			UnderlyingSymbols: underlying,
+			Status:            alpaca.OptionStatusActive,
+			ExpirationDateGTE: dt,
+			StrikePriceGTE:    decimal.NewFromFloat(px + 10), // strike B
+			TotalLimit:        1,
+		}
```

Comment:
Without the {} block this can be simplified to `req.StrikePriceGTE = decimal.NewFromFloat(px + 10), // strike B`

Reply:
7f5ef5cad60686b69fb44fc685de6a282e972c21